### PR TITLE
albert flax

### PR DESCRIPTION
**Pull request by kamalkraj**
# What does this PR do?

ALBERT Flax Model

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@LysandreJik @patrickvonplaten

### CODE REVIEW

**Review by patrickvonplaten — COMMENTED** [review]
Really cool PR @kamalkraj !!!

I think we can merge it soon. The main points to add are:

- A slow integration test making sure that Flax gives the same results as PT for Albert
- Add `# Copied from ...` statements to make sure the code is kept maintainable

**Comment by stefan-it** [comment]
Yeah, it was the TF 1 code base, and I've also trained various model sizes. Let's see if I have more luck using the FLAX implementation 😅

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 1]
```diff
@@ -0,0 +1,1144 @@
+# coding=utf-8
+# Copyright 2018 Google AI, Google Brain and the HuggingFace Inc. team.
```
Comment (patrickvonplaten) [inline src/transformers/models/albert/modeling_flax_albert.py:2]:
```suggestion
# Copyright 2021 Google AI, Google Brain and the HuggingFace Inc. team.
```

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 2]
```diff
@@ -0,0 +1,1144 @@
...
+
+_CHECKPOINT_FOR_DOC = "albert-base-v2"
+_CONFIG_FOR_DOC = "AlbertConfig"
+_TOKENIZER_FOR_DOC = "AlbertTokenizer"
+
+
+@flax.struct.dataclass
+class FlaxAlbertForPreTrainingOutput(ModelOutput):
+    """
+    Output type of :class:`~transformers.AlbertForPreTraining`.
```
Comment (patrickvonplaten) [inline src/transformers/models/albert/modeling_flax_albert.py:59]:
```suggestion
    Output type of :class:`~transformers.FlaxAlbertForPreTraining`.
```

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 3]
```diff
@@ -0,0 +1,1144 @@
...
+
+    Args:
+        prediction_logits (:obj:`torch.FloatTensor` of shape :obj:`(batch_size, sequence_length, config.vocab_size)`):
```
Comment (patrickvonplaten) [inline src/transformers/models/albert/modeling_flax_albert.py:62]:
```suggestion
        prediction_logits (:obj:`jnp.ndarray` of shape :obj:`(batch_size, sequence_length, config.vocab_size)`):
```

Comment (patrickvonplaten) [reply src/transformers/models/albert/modeling_flax_albert.py:62]: and all follow-up occurrences of `torch.FloatTensor`

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 4]
```diff
@@ -0,0 +1,1143 @@
...
+
+        # Sum all embeddings
+        hidden_states = inputs_embeds + token_type_embeddings + position_embeds
+
+        # Layer Norm
+        hidden_states = self.LayerNorm(hidden_states)
+        hidden_states = self.dropout(hidden_states, deterministic=deterministic)
+        return hidden_states
+
+
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply src/transformers/models/albert/modeling_flax_albert.py:187]: Could we try to add `#Copied from ...` statements whereever it's possible :-) Would be important to ensure that the code stays in sync down the road

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 5]
```diff
@@ -0,0 +1,1144 @@
...
+        # make sure `position_ids` is correctly initialized when not passed
+        if position_ids is None:
+            position_ids = jnp.broadcast_to(jnp.arange(jnp.atleast_2d(input_ids).shape[-1]), input_ids.shape)
+
+        hidden_states = self.embeddings(
+            input_ids, token_type_ids, position_ids, attention_mask, deterministic=deterministic
+        )
+
+        if head_mask is not None:
```
Comment (patrickvonplaten) [inline src/transformers/models/albert/modeling_flax_albert.py:644]: Can we remove the `head_mask` completely? It's not yet supported in any Flax model I think

File: src/transformers/models/albert/modeling_flax_albert.py [hunk 6]
```diff
@@ -0,0 +1,1144 @@
...
+    Returns:
+
+    Example::
+
+        >>> from transformers import AlbertTokenizer, FlaxAlbertForPreTraining
+
+        >>> tokenizer = AlbertTokenizer.from_pretrained('albert-base-v2')
+        >>> model = FlaxAlbertForPreTraining.from_pretrained('albert-base-v2')
+
+        >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="jax")
```
Comment (patrickvonplaten) [inline src/transformers/models/albert/modeling_flax_albert.py:770]:
```suggestion
        >>> inputs = tokenizer("Hello, my dog is cute", return_tensors="np")
```
See: https://github.com/huggingface/transformers/pull/13307

File: tests/test_modeling_flax_albert.py [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+            model = model_class_name.from_pretrained("albert-base-v2", from_pt=True)
+            outputs = model(np.ones((1, 1)))
+            self.assertIsNotNone(outputs)
+
+
+@require_flax
+class AlbertModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_no_head_absolute_embedding(self):
+        model = FlaxAlbertModel.from_pretrained("albert-base-v2", from_pt=True)
```
Comment (patrickvonplaten) [inline tests/test_modeling_flax_albert.py:151]:
```suggestion
        model = FlaxAlbertModel.from_pretrained("albert-base-v2")
```

File: tests/test_modeling_flax_albert.py [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+        else ()
+    )
+
+    def setUp(self):
+        self.model_tester = FlaxAlbertModelTester(self)
+
+    @slow
+    def test_model_from_pretrained(self):
+        for model_class_name in self.all_model_classes:
+            model = model_class_name.from_pretrained("albert-base-v2", from_pt=True)
```
Comment (patrickvonplaten) [inline tests/test_modeling_flax_albert.py:142]:
```suggestion
            model = model_class_name.from_pretrained("albert-base-v2")
```

Comment (patrickvonplaten) [reply tests/test_modeling_flax_albert.py:142]: Uploaded the weights :-)

File: tests/test_modeling_flax_albert.py (hunk 7 above)
Comment (patrickvonplaten) [reply tests/test_modeling_flax_albert.py:151]: Uploaded the weights :-)

File: tests/test_modeling_flax_albert.py [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+            outputs = model(np.ones((1, 1)))
+            self.assertIsNotNone(outputs)
+
+
+@require_flax
+class AlbertModelIntegrationTest(unittest.TestCase):
```
Comment (patrickvonplaten) [inline tests/test_modeling_flax_albert.py:148]:
```suggestion
class FlaxAlbertModelIntegrationTest(unittest.TestCase):
```

**Comment by patrickvonplaten** [comment]
Awesome addition @kamalkraj - thanks a lot :-) Left a couple of final additions

File: src/transformers/models/albert/modeling_flax_albert.py (hunk 1 above)
Comment (kamalkraj) [reply src/transformers/models/albert/modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/b7ad627746d90bfd81974a8393f936a2332a0454

File: src/transformers/models/albert/modeling_flax_albert.py (hunk 2 above)
Comment (kamalkraj) [reply src/transformers/models/albert/modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/7cbf87e351d5c0cef5c5996b11f6f8d348d994c4

File: src/transformers/models/albert/modeling_flax_albert.py (hunk 3 above)
Comment (kamalkraj) [reply src/transformers/models/albert/modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/7cbf87e351d5c0cef5c5996b11f6f8d348d994c4

File: src/transformers/models/albert/modeling_flax_albert.py (hunk 5 above)
Comment (kamalkraj) [reply src/transformers/models/albert/modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/723d9f49eac97ebac3508e4b03c6c34e0fd2540f

File: src/transformers/models/albert/modeling_flax_albert.py (hunk 6 above)
Comment (kamalkraj) [reply src/transformers/models/albert/modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/7cbf87e351d5c0cef5c5996b11f6f8d348d994c4

File: tests/test_modeling_flax_albert.py (hunk 7 above)
Comment (kamalkraj) [reply tests/test_modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/b303e119ce625120d49faeaec796af07908f4ecd

File: tests/test_modeling_flax_albert.py (hunk 8 above)
Comment (kamalkraj) [reply tests/test_modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/b303e119ce625120d49faeaec796af07908f4ecd

File: tests/test_modeling_flax_albert.py (hunk 9 above)
Comment (kamalkraj) [reply tests/test_modeling_flax_albert.py]: done.
https://github.com/huggingface/transformers/pull/13294/commits/b303e119ce625120d49faeaec796af07908f4ecd

**Comment by kamalkraj** [comment]
@patrickvonplaten 
Done changes according to review.

**Comment by patrickvonplaten** [comment]
Slow tests are passing on CPU - thanks for the model addition @kamalkraj !

**Pull request merged**